### DISCUSSION TITLE
vecs create_index() timeout.

### DISCUSSION THREAD

**Discussion by Sangeeth-versos** [supabase/supabase#14622] [Questions]
create_index()  works fine when I have less data in the collection. But when I have large data, I am getting the following error.

sqlalchemy.exc.OperationalError: (psycopg2.errors.QueryCanceled) canceling statement due to statement timeout. 

The time required to create an index grows with the number of records and size of vectors has been mentioned in the document. But how to handle the  timeout?

**Comment by louis030195** [comment]
same
i am not using vecs, just pgvector and impossible to index my 600.000 rows even using direct sql connection
